semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -1316,10 +1316,17 @@ and m_arguments_concat a b =
 
   (*e: [[Generic_vs_generic.m_arguments_concat()]] ellipsis cases *)
   (* the general case *)
-  | xa::aas, xb::bbs ->
-      m_argument xa xb >>= (fun () ->
-        m_arguments_concat aas bbs
-      )
+  | xa::aas, xb::bbs -> (
+      (* exception: for concat strings, don't have ellipsis match   *)
+      (* string literals since string literals are implicitly not   *)
+      (* interpolated, and ellipsis implicitly is                   *)
+      match xa, xb with
+      | A.Arg (A.Ellipsis _), A.Arg (A.L _) ->
```

Comment:
hmm, maybe be even more explicit and rquire A.L (A.String here?) instead of just any literal?